### ISSUE TITLE
webpack resolve modules jail functionality

### ISSUE THREAD

**Issue by Globegitter** [webpack/webpack#9165]
**What is the expected behavior?**

When providing `resolve.modules = ['node_modules']` webpack will potentially traverse all the way up to the root directory if it does not find the module. I should be able to add an option, maybe `resolve/resolveLoader.jail = '/my/jail/path` the way rollup allows: https://github.com/rollup/rollup-plugin-node-resolve#usage so the traversels can be limited to stay within a directory and its subdirectories. Alternatively I could also imagine having a `resolve.traverse = False` setting so relative paths do not have the traversal behaviour similar to absolute paths.

**What is motivation or use case for adding/changing the behavior?**

When using a build tool like [Bazel](https://bazel.build/) that tries to isolate an application as much as possible this is a no-go as adding a node_modules in a completely different directory can change the behaviour of the application (and for a tool like bazel one would rather want the application to fail than it finding some stray modules). Only using absolute paths does unfortunately not work either as then it does not find nested `node_modules` of dependencies.

Also it would be nice to have parity with rollup.

**How should this be implemented in your opinion?**

As mentioned, one of two config options would be good:

```
resolve/resolveLoader.jail = '/my/jail/path'
```
or

```
resolve/resolveLoader.traverse = False
```

The first one would simply make the traversal stop at the given path and the second option would simply allow for now upwards traversal for relative paths.

**Are you willing to work on this yourself?**
If I get some pointers on where/how to implemented this I would be happy to give it a try myseld.

**Comment by sokra** [comment]
> If I get some pointers on where/how to implemented this I would be happy to give it a try myseld.

https://github.com/webpack/enhanced-resolve

Send a PR

**Comment by evilebottnawi** [comment]
bump

**Comment by Globegitter** [comment]
@evilebottnawi there is a PR that adds the functionality: https://github.com/webpack/enhanced-resolve/pull/175 but still waiting to hear back from @sokra

**Comment by sokra** [comment]
fixed by webpack/enhanced-resolve#221
